tests/test_download_photos.py

```diff
@@ -1506,6 +1506,61 @@ def test_download_after_delete(self):
         for file_name in files_to_download:
             assert os.path.exists(os.path.join(base_dir, os.path.normpath(file_name))), f"File {file_name} expected, but does not exist"
 
+    def test_download_after_delete_fail(self):
+        base_dir = os.path.normpath(f"tests/fixtures/Photos/{inspect.stack()[0][3]}")
+        if os.path.exists(base_dir):
+            shutil.rmtree(base_dir)
+        os.makedirs(base_dir)
+
+        with mock.patch("icloudpd.download.download_media") as dp_patched:
+            dp_patched.return_value = False
+            with vcr.use_cassette("tests/vcr_cassettes/listing_photos.yml"):
+                # Pass fixed client ID via environment variable
+                runner = CliRunner(env={
+                    "CLIENT_ID": "DE309E26-942E-11E8-92F5-14109FE0B321"
+                })
+                result = runner.invoke(
+                    main,
+                    [
+                        "--username",
+                        "jdoe@gmail.com",
+                        "--password",
+                        "password1",
+                        "--recent",
+                        "1",
+                        "--skip-videos",
+                        "--skip-live-photos",
+                        "--no-progress-bar",
+                        "--threads-num",
+                        1,
+                        "--delete-after-download",
+                        "-d",
+                        base_dir,
+                    ],
+                )
+                print_result_exception(result)
+
+                self.assertIn("DEBUG    Looking up all photos from album All Photos...", self._caplog.text)
+                self.assertIn(
+                    f"INFO     Downloading the first original photo to {base_dir} ...",
+                    self._caplog.text,
+                )
+                self.assertIn(
+                    f"INFO     Downloading {os.path.join(base_dir, os.path.normpath('2018/07/31/IMG_7409.JPG'))}",
+                    self._caplog.text,
+                )
+                self.assertNotIn(
```

Comment:
Is there a way to confirm skipping deletion call without/in addition to relying on logs? It would be beneficial to separate logic check from logging check, so we can change one without breaking tests for the other.

Note the most test still combine log and state/file/api checks and I hope that can be changes over time too.

Reply:
I wanted to mock the internal `delete_photo()` function to see if it was called, but it seems it's not accessible. The best way I found to make that tests is to craft a cassette to simulate a bad request. Because of that bad request, the delete is never called so the cassette is complete.